crates/ruff_python_formatter/src/expression/expr_attribute.rs

```diff
@@ -179,7 +179,11 @@ impl NeedsParentheses for ExprAttribute {
             context.comments().ranges(),
             context.source(),
         ) {
-            OptionalParentheses::Never
+            if context.comments().has_trailing(self.value.as_ref()) {
```

Comment:
We'll now also use the new layout if `value` is parenthesized and has a trailing comment like this

```python
variable = (
    (something # a comment  
    ).fist_method("some string")
)
```

I think that's fine but it might be worth adding a few more tests